src/bridges/set/IssuanceBridge.sol

```diff
@@ -49,150 +71,83 @@ contract IssuanceBridge is IDefiBridge {
-        // inputAssetA: SetToken
-        // outputAssetA ERC20 or ETH
-        if (
-            setController.isSet(address(inputAssetA.erc20Address)) &&
-            (outputAssetA.assetType == AztecTypes.AztecAssetType.ETH ||
-                outputAssetA.assetType == AztecTypes.AztecAssetType.ERC20)
-        ) {
-            // Check that spending of the given SetToken is approved
-            require(
-                IERC20(inputAssetA.erc20Address).approve(
-                    address(exchangeIssuance),
-                    inputValue
-                ),
-                "IssuanceBridge: APPROVE_FAILED"
-            );
+        if (msg.sender != rollupProcessor) revert InvalidCaller();
+        if (inputValue == 0) revert ZeroInputValue();
 
-            // user wants to receive ETH
+        if (SET_CONTROLLER.isSet(address(inputAssetA.erc20Address))) {
+            // User wants to redeem the underlying asset
             if (outputAssetA.assetType == AztecTypes.AztecAssetType.ETH) {
-                outputValueA = exchangeIssuance.redeemExactSetForETH(
+                // redeem in exchange for ETH
+                outputValueA = EXCHANGE_ISSUANCE.redeemExactSetForETH(
                     ISetToken(address(inputAssetA.erc20Address)),
                     inputValue, // _amountSetToken
-                    0 // __minEthOut
+                    auxData // __minEthOut
                 );
-                // Send ETH to rollup processor
                 IRollupProcessor(rollupProcessor).receiveEthFromBridge{value: outputValueA}(interactionNonce);
-            }
-
-            // user wants to receive ERC20
-            if (outputAssetA.assetType == AztecTypes.AztecAssetType.ERC20) {
-                outputValueA = exchangeIssuance.redeemExactSetForToken(
+            } else if (outputAssetA.assetType == AztecTypes.AztecAssetType.ERC20) {
+                // redeem in exchange for ERC20
+                outputValueA = EXCHANGE_ISSUANCE.redeemExactSetForToken(
                     ISetToken(address(inputAssetA.erc20Address)), // _setToken,
                     IERC20(address(outputAssetA.erc20Address)), // _outputToken,
                     inputValue, //  _amountSetToken,
-                    0 //  _minOutputReceive
+                    auxData //  _minOutputReceive
                 );
-
-                // approve the transfer of funds back to the rollup contract
-                require(
-                    IERC20(outputAssetA.erc20Address).approve(
-                        rollupProcessor,
-                        outputValueA
-                    ),
-                    "IssuanceBridge: APPROVE_FAILED"
+            } else {
+                revert IncorrectInput();
+            }
+        } else if (SET_CONTROLLER.isSet(address(outputAssetA.erc20Address))) {
+            // User wants to issue SetToken
+            if (inputAssetA.assetType == AztecTypes.AztecAssetType.ERC20) {
```

Comment:
Think it would help a tiny bit on readability to keep it in the same ordering, e.g., ETH first then the non-eth.

Reply:
Good point 👍